### PR TITLE
(#3797) - consistently use utils.btoa/utils.atob

### PR DESCRIPTION
**Pull request by nolanlawson**
Also move them to their own file outside of utils.
This is just code cleanup and doesn't change the
functionality at all.

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -248,31 +247,9 @@ Changes.prototype.notify = function (dbName) {
...
-    return btoa(str);
-  };
-} else {
-  exports.btoa = function (str) {
-    return new buffer(str, 'binary').toString('base64');
-  };
-}
+var base64 = require('./deps/base64');
+exports.atob = base64.atob;
+exports.btoa = base64.btoa;
```
Comment (daleharvey) [inline lib/utils.js]: These are only called in 6 places, before merging fancy getting rid of them from utils at all and requiring base64 instead?

**Comment by daleharvey** [comment]
:+1: I think we can clean it up a bit further but either way can merge, cheers

File: lib/utils.js (hunk 1 above)
Comment (nolanlawson) [inline lib/utils.js]: yeah, I like that idea. will do

**Comment by nolanlawson** [comment]
Ack, well, I can't seem to get the tests to pass; we're using `utils.btoa` and `utils.atob` in tons of places in the tests. I'll keep working on it later, but opened up a separate WIP PR for now.

**Comment by daleharvey** [comment]
Heh sorry I only grepped ./lib and thought it was only in a few places

**Comment by nolanlawson** [comment]
Merging this PR; as I said, the other one was a bit tricker.

**Comment by nolanlawson** [comment]
06e84f266f387ef351501ec777b8916e7d88fd77

**Pull request closed without merging**